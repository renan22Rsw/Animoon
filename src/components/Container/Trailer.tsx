import SubTitles from "../Titles/SubTitles";

const Trailer = ({ trailerId }: { trailerId: string }) => {
  return (
    <>
      <SubTitles>Trailer</SubTitles>

      <iframe
        className=" w-[420px] h-[300px] p-10 md:w-[500px] md:h-[300px] md:p-4 "
        src={`https://www.youtube.com/embed/${trailerId}?enablejsapi=1&wmode=opaque&autoplay=0`}
        title="YouTube video player"
        allow="accelerometer; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
        referrerPolicy="strict-origin-when-cross-origin"
        allowFullScreen
      ></iframe>
    </>
  );
};

export default Trailer;
