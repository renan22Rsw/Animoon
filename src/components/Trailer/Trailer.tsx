import SubTitles from "../Titles/SubTitles";

const Trailer = ({ trailerId }: { trailerId: string }) => {
  return (
    <>
      <SubTitles>Trailer</SubTitles>

      <iframe
        className=" w-full h-[300px] p-6  md:w-[500px] md:h-[300px] md:p-4 "
        src={`https://www.youtube.com/embed/${trailerId}?enablejsapi=1&wmode=opaque&autoplay=0`}
        title="YouTube video player"
        allow="accelerometer; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
        referrerPolicy="strict-origin-when-cross-origin"
        allowFullScreen
        data-testid="iframe"
      ></iframe>
    </>
  );
};

export default Trailer;
