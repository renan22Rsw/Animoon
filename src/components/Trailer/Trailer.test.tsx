import { render, screen } from "@testing-library/react";
import Trailer from "./Trailer";

const id = "abc";

describe("Trailer component", () => {
  it("should render SubTitles from Trailer", () => {
    render(<Trailer trailerId={id} />);
    expect(screen.getByText("Trailer")).toBeInTheDocument();
  });

  it("should render Iframe", () => {
    render(<Trailer trailerId={id} />);
    expect(screen.getByTestId("iframe")).toBeInTheDocument();
    expect(screen.getByTestId("iframe")).toHaveAttribute(
      "src",
      `https://www.youtube.com/embed/${id}?enablejsapi=1&wmode=opaque&autoplay=0`
    );
  });

  it("should match a snapshot", () => {
    const { container } = render(<Trailer trailerId={id} />);
    expect(container).toMatchSnapshot();
  });
});
