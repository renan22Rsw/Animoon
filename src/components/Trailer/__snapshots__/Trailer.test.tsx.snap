// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trailer component should match a snapshot 1`] = `
<div>
  <div
    class="m-4 px-4 text-[#EDEDED]"
  >
    Trailer
  </div>
  <iframe
    allow="accelerometer; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
    allowfullscreen=""
    class=" w-full h-[300px] p-6  md:w-[500px] md:h-[300px] md:p-4 "
    data-testid="iframe"
    referrerpolicy="strict-origin-when-cross-origin"
    src="https://www.youtube.com/embed/abc?enablejsapi=1&wmode=opaque&autoplay=0"
    title="YouTube video player"
  />
</div>
`;
